File: sem4/src/CashProvider.java
public class CashProvider {
    public long card;
    public long hashCard;
    public boolean isAutorization = false;

    public void authorization() {
        isAutorization = true;
    }

    public boolean buy(Ticket ticket) {
        if (isAutorization) {
            return true;
        }
        return false;
    }




}

File: sem4 py/BuyTicket/CashProvider.py
class CashProvider:
    def __init__(self, havemoney: int = 0):
        self.haveMoney = havemoney
        self.isAutorization = isAutorization = False

    def AutoRiz(self):
        return True

    def Buy(self, price: int) -> bool:
        if self.haveMoney >= price:
            return True
